### ISSUE TITLE
DSL for Http Server to assert on Webhooks and similar

### ISSUE THREAD

**Issue by cneijenhuis** [agourlay/cornichon#83]
I want to test that my API is calling webhooks, and that it is sending the data as specified. Currently, I need to write my own HTTP server that saves and returns the data, so I can assert it. It would be awesome to have this integrated in Cornichon.

Here's an example with a made-up DSL:

```scala
// Create the HTTP server that listens for webhooks
When I httpListenTo("http://localhost:12345/webhook")
// Do something that should trigger the webhook
When I post("/users").withBody(
"""
{
  "firstname": "Foo",
  "lastname": "Bar"
}
""")
Then assert status.is(201)
// Check that webhook was delivered
Then assert httpListen.received_calls(1)
And assert httpListen.lastBody.is(...)
```

**Comment by agourlay** [comment]
Here is what I believe I can propose for a start.

```scala
httpListenTo(label = "my-server", port = 12345){
  When I post("/users").withBody(
  """
  {
    "firstname": "Foo",
    "lastname": "Bar"
  }
  """) 
  Then assert status.is(201)
}
// Check that webhook was delivered
Then assert httpListen("my-server").received_calls(1)
And assert httpListen("my-server").receivedRequest(1).onUrl("...").withVerb("...").withBody(...)
```

Design points:
- the server is automatically started and stopped at the end of the block.
- assertions on the server must be done outside of the block.
- by default POST will receive 201 and the REST 200.

WDYT?

**Comment by cneijenhuis** [comment]
Sounds great! 👍

**Comment by agourlay** [comment]
0.9.1 released with experimental HTTP mock feature

some usage example : https://github.com/agourlay/cornichon/blob/master/src/test/scala/com/github/agourlay/cornichon/examples/MockServerExample.scala

**Comment by agourlay** [comment]
Several person asked me about having the possibility to define the behaviour of the mock server.

It could be possible to offer an entry point in the DSL to inject the definition of an akka-http route.

I will think about it for a while.

**Comment by agourlay** [comment]
The design had to evolved to support a new feature. 
The HTTP mock now exposes all the requests it recorded through an HTTP API

**Comment by agourlay** [comment]
first public version shipped in 0.9.3 + minimal doc available in Readme.
